### PR TITLE
Transform

### PR DESCRIPTION
**Pull request by DKWeeto**
transform script, including transform and load flood warnings

### CODE REVIEW

File: etl_pipeline/transform.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        concentration = 'error'
+    severity_id = calculate_air_quality_alert(concentration)
+    return {'o3_concentration': concentration, 'severity_id': severity_id}
+
+
+if __name__ == "__main__":
+    load_dotenv()
+    for x in gather_weather_data(0, 0):
+        print(x)
+    print(gather_air_quality(0, 0))
```
Comment (DOM-C1) [inline etl_pipeline/transform.py]: do we need to print these or just a debugging artifact?

File: ETL_pipeline/transform_flood.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    flood_warnings = get_location_columns(flood_warnings)
+    flood_warnings = flood_warnings[['latitude', 'longitude',
+                                     'severity_level_id',
+                                     'time_raised']].drop_duplicates(ignore_index=True)
+    flood_warnings = flood_warnings.reset_index(drop=True)
+    return flood_warnings.to_dict(orient="records")
+
+
+if __name__ == "__main__":
+    print(get_all_floods())
```
Comment (DOM-C1) [inline ETL_pipeline/transform_flood.py]: is there a reason we are printing the get_all_floods()?

File: database/insert_metadata.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 
 def get_db_connection(config):
-    """Connect to the database"""
+    """Connect to the database."""
```
Comment (DOM-C1) [inline database/insert_metadata.py:11]: nice use of punctuation

File: etl_pipeline/transform.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+
+def calculate_visibility_alerts(visibility: pd.Series) -> int:
+    """Find low visibility alerts amongst visibility data."""
+    if 20 >= visibility:
+        return 1
+    if 50 >= visibility > 20:
+        return 2
+    if 150 >= visibility > 0.5:
+        return 3
```
Comment (DOM-C1) [inline etl_pipeline/transform.py]: Looking good, small suggestion, I think it's a bit more readable to have the largest element on the right hand-side and the smallest element on the left hand side of the inequality so 150 >= visibility > 0.5   would become 0.5 < visibility <= 150.

File: etl_pipeline/transform.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    return 4
+
+
+def calculate_air_quality_alert(concentration: float) -> int:
+    """Find if air quality is an alert from the concentration of o3."""
+    if 0 <= concentration < 101:
+        severity_id = 4
+    elif 101 <= concentration < 161:
+        severity_id = 3
+    elif 161 <= concentration < 241:
```
Comment (DOM-C1) [inline etl_pipeline/transform.py]: this time you do put the largest on the right hand side. please make standard notation instead of switching styles.

**Review by DOM-C1 — COMMENTED** [review]
Looking good

File: ETL_pipeline/transform_flood.py (hunk 2 above)
Comment (NMcKittie) [reply ETL_pipeline/transform_flood.py]: I assume thats for manually testing

File: database/insert_metadata.py (hunk 3 above)
Comment (DKWeeto) [reply database/insert_metadata.py:11]: thanks

File: etl_pipeline/transform.py (hunk 4 above)
Comment (DKWeeto) [reply etl_pipeline/transform.py]: fixed

File: ETL_pipeline/load_flood.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
+"""Load flood warnings into the database"""
+
+from os import environ as ENV
+
+import psycopg2
```
Comment (arj0713) [inline ETL_pipeline/load_flood.py]: Can we only import what we need, `from ... import ...`

File: ETL_pipeline/load_flood.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+import psycopg2.extras
```
Comment (arj0713) [inline ETL_pipeline/load_flood.py]: Might have missed something but is this used in this file?

File: ETL_pipeline/load_flood.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+                        AND loc_id = {loc_id}""")
+            flood = cur.fetchone()
+            if not flood:
+                cur.execute(f"""INSERT INTO flood_warnings
+                            (severity_level_id, time_raised, loc_id, notified)
+                            VALUES ({severity_level_id}, '{time_raised}', {loc_id}, False)""")
+                conn.commit()
+
+
+def insert_all_floods(config, floods: list[dict]) -> None:
```
Comment (arj0713) [inline ETL_pipeline/load_flood.py]: `config` type hint?

File: etl_pipeline/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+geopy
+pandas
+psycopg2-binary
+pytest
```
Comment (arj0713) [inline etl_pipeline/requirements.txt]: Do you need pytest to run the files?

File: ETL_pipeline/transform_flood.py (hunk 2 above)
Comment (DKWeeto) [reply ETL_pipeline/transform_flood.py]: removed

File: etl_pipeline/transform.py (hunk 1 above)
Comment (DKWeeto) [reply etl_pipeline/transform.py]: removed

File: etl_pipeline/transform.py (hunk 5 above)
Comment (DKWeeto) [reply etl_pipeline/transform.py]: did it

File: etl_pipeline/requirements.txt (hunk 9 above)
Comment (DKWeeto) [reply etl_pipeline/requirements.txt]: no i guess not

File: ETL_pipeline/load_flood.py (hunk 8 above)
Comment (DKWeeto) [reply ETL_pipeline/load_flood.py]: what IS the typehint for config/environ?

File: ETL_pipeline/load_flood.py (hunk 7 above)
Comment (DKWeeto) [reply ETL_pipeline/load_flood.py]: i forgot to include cursorfactory

File: ETL_pipeline/load_flood.py (hunk 6 above)
Comment (DKWeeto) [reply ETL_pipeline/load_flood.py]: yep

File: ETL_pipeline/load_flood.py (hunk 8 above)
Comment (DKWeeto) [reply ETL_pipeline/load_flood.py]: os._Environ ?

Comment (DKWeeto) [reply ETL_pipeline/load_flood.py]: ok i did dict

File: ETL_pipeline/load_flood.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+def get_county(address: dict) -> str | None:
+    """Extract the county name of a location."""
+    if 'county' in address:
+        return address['county']
+    if 'state_district' in address:
+        return address['state_district']
+    return None
+
+
+def get_country(address: dict) -> str | None:
```
Comment (NMcKittie) [inline ETL_pipeline/load_flood.py]: you shouldnt return 2 types

**Review by NMcKittie — APPROVED** [review]
Looks Good

**Pull request merged**